src/pydocstyle/parser.py

```diff
@@ -267,7 +267,27 @@ def move(self):
         current = self._next_from_generator()
         self.current = None if current is None else Token(*current)
         self.line = self.current.start[0] if self.current else self.line
-        self.got_logical_newline = (previous.kind in self.LOGICAL_NEWLINES)
+        is_logical_blank = previous.kind in (tk.NL, tk.COMMENT)
+        # Originally the concept of logical_new_line was introduced
+        # to ensure that we were able to differentiate between
+        # the matmul operator ("@") and a decorator.
+        # In case we didn't have a "logical new line", it meant
+        # we were parsing a matmul operator instead of a decorator.
+        # Eg -
+        # A @ B
+        # def test():...
+        # v/s
+        # @decorator
+        # def test(): ...
+        self.got_logical_newline = (
+            previous.kind in self.LOGICAL_NEWLINES
+            # In case we had a logical new line previously and
+            # the previous token was a new line we should persist this state.
+            # This is so we don't skip parsing trivial decorator
+            # cases like as described
+            # in https://github.com/PyCQA/pydocstyle/issues/410
```

Comment:
This can also be more concise, e.g.:

```
self.got_logical_newline = (
    previous.kind in self.LOGICAL_NEWLINES
    # Retain logical_newline status if last line was logically blank
    or (self.got_logical_newline and previous.kind in self.LOGICAL_BLANKS)
)
```